pixeltable/utils/pydantic_serializable.py

```diff
@@ -0,0 +1,71 @@
+"""
+Pydantic serialization for Pixeltable classes.
+
+This base class provides automatic Pydantic serialization/deserialization using the as_dict() and from_dict() methods.
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, TypeVar
+
+from pydantic_core import CoreSchema, core_schema
+
+# TypeVar for generic return type in from_dict() to ensure that subclasses return their own type
+T = TypeVar('T', bound='PydanticSerializable')
+
+
+class PydanticSerializable(ABC):
+    """
+    Abstract base class that provides Pydantic core schema using as_dict() and from_dict() methods.
+    """
+
+    @abstractmethod
+    def as_dict(self) -> dict[str, Any]:
+        """Convert object to dictionary representation."""
+        pass
+
+    @classmethod
+    @abstractmethod
+    def from_dict(cls: type[T], data: dict[str, Any]) -> T:
+        """Create object from dictionary representation."""
+        pass
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls,
+        source_type: Any,
+        handler: Any,  # GetCoreSchemaHandler
+    ) -> CoreSchema:
+        """
+        Provide Pydantic core schema for serialization/deserialization.
+
+        Args:
+            source_type: The source type being processed
+            handler: Pydantic's schema handler
+
+        Returns:
+            CoreSchema that uses as_dict() and from_dict() for serialization
+        """
+
+        def deserialize_from_dict(data: Any) -> Any:
```

Comment:
Is it possible to be more type-friendly here?
```python
def deserialize_from_dict(data: Self | dict) -> Self:
```
or will that cause problems with the Pydantic interface?